types/object.go

```diff
@@ -101,6 +101,16 @@ func (o ObjectType) Equal(candidate attr.Type) bool {
 	return true
 }
 
+// ApplyTerraform5AttributePathStep applies the given AttributePathStep to the
+// object.
+func (o ObjectType) ApplyTerraform5AttributePathStep(step tftypes.AttributePathStep) (interface{}, error) {
+	if _, ok := step.(tftypes.ElementKeyString); !ok {
```

Comment:
I think this wants to be an AttributeName.
```suggestion
	if _, ok := step.(tftypes.AttributeName); !ok {
```

Because objects have attributes, not elements.

Reply:
Thanks, adding a test